frontend/src/core/api/base.js

```diff
@@ -87,4 +87,39 @@ export default class APIBase {
             return {};
         }
     }
+
+    keysToCamelCase(results) {
+        const toCamelCase = (s) => {
+            return s.replace(/([-_][a-z])/ig, ($1) => {
+              return $1.toUpperCase()
+                .replace('-', '')
+                .replace('_', '');
+            });
+        };
+
+        const isArray = function (arr) {
+            return Array.isArray(arr);
+        };
+
+        const isObject = function (obj) {
+            return obj === Object(obj) && !isArray(obj) && typeof obj !== 'function';
```

Comment:
I think the `isArray` function is superfluous, we can do without it. You can simply use `Array.isArray` here instead.